src/drivers/fluid_jack.c

```diff
@@ -445,9 +455,26 @@ fluid_jack_client_register_ports(void *driver, int isaudio, jack_client_t *clien
         }
 
         fluid_settings_getint(settings, "synth.effects-channels", &dev->num_fx_ports);
+
+        if(synth->effects_channels != dev->num_fx_ports)
+        {
+            FLUID_LOG(FLUID_ERR, "synth is assuming %d effect channels, while jack is using %d",
+                      synth->effects_channels * synth->effects_groups,
+                      dev->num_fx_ports);
+            goto error_recovery;
+        }
+
         fluid_settings_getint(settings, "synth.effects-groups", &i);
 
         dev->num_fx_ports *= i;
+
+        if(synth->effects_groups != i)
```

Comment:
Possible NULL deref on `synth`

Reply:
I updated the PR with the above changes. Yet, if you prefer, I can try to come up with a different solution for issue #724, as discussed there.